pkgs/development/interpreters/perl/default.nix

```diff
@@ -72,6 +72,13 @@ let
       (if crossCompiling
        then [ "-Dlibpth=\"\"" "-Dglibpth=\"\"" ]
        else [ "-de" "-Dcc=cc" ])
+
+      # prevent taking architecture of the builder machine which could be "aarch64"
+      # it should be safe to have always enabled, (stdenv.isAarch32 && stdenv.isLinux) here is to prefend mass-rebuild
+      ++ optionals (stdenv.isAarch32 && stdenv.isLinux && !crossCompiling) [
```

Comment:
Does ```!crossCompiling``` need to be here?

Reply:
for not to break cross-compiling (I have not tested it).
I guess it will be safe to keep here only ```stdenv.isLinux```, but it requires testing